File: 정보올림피아드_초등부/2635_수 이어가기.py
n = int(input())

#최대 길이, 최대길이 리스트
max_len = 0
max_arr = 0

for i in range(n+1):
    result = [n,i]
    j=0
    
    while(True):
        #해당값 조회
        ele = result[j] - result[j+1]
        
        #음수가 등장하면
        if ele <= -1:
            #탈출
            break
        
        #삽입
        result.append(ele)

        #최대인지 체크
        if max_len < len(result):
            max_len = len(result)
            max_arr = result
        
        #증가
        j += 1

#출력
print(max_len)
print(*max_arr)

File: 정보올림피아드_초등부/2635_수 이어가기.cpp
#include <iostream>
#include <vector>

using namespace std;

void solution(void) {
	int n;
	int max_len = 0;
	vector<int> max_arr;

	cin >> n;
	
	for (int i = 0; i <= n; ++i) {
		vector<int> arr;
		arr.push_back(n);
		arr.push_back(i);
		int idx = 0;

		while (true) {
			int ele = arr[idx] - arr[idx + 1];

			if (ele <= -1)
				break;

			arr.push_back(ele);

			if (max_len < arr.size()) {
				max_len = arr.size();
				max_arr = arr;
			}
			idx++;
		}
	}

	cout << max_len << '\n';
	for (int i = 0; i < max_len; ++i) 
		cout << max_arr[i] << ' ';
	cout << '\n';
}

int main(void) {
	//ÃÊ±âÈ­
	ios::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	solution();
	return 0;
}